File: 27_spring_cache/local_cache/src/main/java/com/yologger/local_cache/service/MemberDTO.java
package com.yologger.local_cache.service;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.yologger.local_cache.repository.MemberEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Builder
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class MemberDTO implements Serializable {

    @JsonProperty("id") private Long id;
    @JsonProperty("email") private String email;
    @JsonProperty("password") private String password;

    public MemberEntity toEntity() {
        return MemberEntity.builder()
                .email(this.email)
                .password(this.password)
                .build();
    }
}

File: 14_spring_webflux/02_webflux_kotlin/webflux_kotlin/src/main/kotlin/com/yologger/samples/webflux_kotlin/service/MemberDTO.kt
package com.yologger.samples.webflux_kotlin.service

import com.yologger.samples.webflux_kotlin.repository.MemberEntity

data class MemberDTO(
    val id: Long? = null,
    val email: String,
    val name: String,
    val password: String
) {
    fun toEntity(): MemberEntity {
        return MemberEntity(id = id, email = email, name = name, password = password)
    }
}